lib/intercom-rails/proxy/user.rb

```diff
@@ -30,7 +30,7 @@ def self.current_in_context(search_object)
         potential_user_objects.each do |potential_object|
           begin
             user_proxy = new(search_object.instance_eval(&potential_object), search_object)
-            return user_proxy if user_proxy.valid?
+            return user_proxy if user_proxy.valid? || config.user.as_lead
```

Comment:
This method checks all `potential_user_objects` to try to find a logged in user. If we want to introduce lead-specific behaviour, we should do it afterwards, outside the loop.

If an app stores the currently logged in user as `@user` and enables this behaviour, then all of their logged in users will be sent to Intercom as leads. The first iteration of the loop will check `current_user` and get `nil`, but then return the empty user proxy anyway.